integrations/cats/src/main/scala/sttp/tapir/codec/cats/TapirCodecCats.scala

```diff
@@ -27,12 +28,18 @@ trait TapirCodecCats {
   implicit def schemaForNes[T: Schema]: Schema[NonEmptySet[T]] =
     Schema[NonEmptySet[T]](SchemaType.SArray(implicitly[Schema[T]])).copy(isOptional = false)
 
-  implicit def codecForNonEmptyList[L, H, CF <: CodecFormat](implicit c: Codec[L, List[H], CF]): Codec[L, NonEmptyList[H], CF] =
-    c.mapDecode { l => DecodeResult.fromOption(NonEmptyList.fromList(l)) }(_.toList)
+  implicit def codecForNonEmptyList[L, H: Validator, CF <: CodecFormat](implicit c: Codec[L, List[H], CF]): Codec[L, NonEmptyList[H], CF] =
+    c.modifySchema(_.copy(isOptional = false))
+      .validate(nonEmptyValidator[H, List])
```

Comment:
won't this add element validation twice? The base codec `c` should already validate the list elements (`H`). The only additional validation to add here, would be the `minSize`.

This is distinct from finding a validator for the entire structure, which needs to validate both the elements and the structure.

Reply:
@adamw yes you're right.